modules/internal/ChapelArray.chpl

```diff
@@ -593,7 +593,17 @@ module ChapelArray {
   // Support for domain expressions, e.g., {1..3, 1..3}
   //
 
-  proc chpl__buildDomainExpr(ranges: range(?) ...?rank) {
+  proc chpl__isTupleOfRanges(tup) param {
+    for param i in 1..tup.size {
+      if !isRangeType(tup(i).type) then
+        return false;
+    }
+    return true;
+  }
+
+  proc chpl__buildDomainExpr(ranges...)
+  where chpl__isTupleOfRanges(ranges) {
```

Comment:
This feels unfortunate to me. I see how ``range(?t) ...`` could require all elements to have the same type, but could we allow the old behavior for ``range(?)...``?

Reply:
(Off-list you @benharsh made it clear this wasn't meant to be a request for this PR). Even so, I think I'd rather have the language be more consistent in this area than have a convenience for `range(?)`.